smithy4s-client/src/main/scala/kinesis4cats/smithy4s/client/middleware/RequestLogger.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2023-2023 etspaceman
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kinesis4cats.smithy4s.client
+package middleware
+
+import cats.effect.syntax.all._
+import cats.effect.{Async, Resource}
+import cats.syntax.all._
+import fs2.{Chunk, Pipe, Stream}
+import org.http4s.Response
+import org.http4s.client.Client
+import org.typelevel.log4cats.StructuredLogger
+
+import kinesis4cats.logging.LogContext
+
+object RequestLogger {
```

Comment:
Mostly copied this from http4s. The http4s implementation records the requests and responses as a part of the message. This one will encode them as a part of the context, per the structured logging desires.